app/lib/features/tasks/sheets/create_update_task_item.dart

```diff
@@ -0,0 +1,228 @@
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).taskName,
+          style: Theme.of(context).textTheme.bodySmall,
+        ),
+        const SizedBox(height: 5),
+        TextFormField(
+          autofocus: true,
+          decoration: InputDecoration(
+            hintText: L10n.of(context).name,
+          ),
+          autovalidateMode: AutovalidateMode.onUserInteraction,
+          controller: _taskNameController,
+          validator: (value) => (value?.isNotEmpty == true)
+              ? null
+              : L10n.of(context).pleaseEnterAName,
+        ),
+      ],
+    );
+  }
+
+  Widget _widgetDescriptionName() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).description,
+          style: Theme.of(context).textTheme.bodySmall,
+        ),
+        const SizedBox(height: 5),
+        TextFormField(
+          decoration: InputDecoration(
+            hintText: L10n.of(context).description,
+          ),
+          minLines: 4,
+          maxLines: 4,
+          controller: _taskDescriptionController,
+        ),
+      ],
+    );
+  }
+
+  Widget _widgetDueDate() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          L10n.of(context).dueDate,
+          style: Theme.of(context).textTheme.bodySmall,
+        ),
+        const SizedBox(height: 5),
+        TextFormField(
+          readOnly: true,
+          decoration: InputDecoration(
+            hintText: L10n.of(context).dueDate,
+            suffixIcon: IconButton(
+              onPressed: selectDueDate,
+              icon: const Icon(Icons.calendar_month),
+            ),
+          ),
+          onTap: selectDueDate,
+          controller: _taskDueDateController,
+        ),
+      ],
+    );
+  }
+
+  Future<void> selectDueDate() async {
+    final date = await showDatePicker(
+      context: context,
+      initialDate: selectedDate,
+      firstDate: DateTime.now(),
+      lastDate: DateTime.now().addYears(1),
+    );
+    if (date == null || !mounted) return;
+    selectedDate = date;
+    _taskDueDateController.text = taskDueDateFormat(date);
+    setState(() {});
+  }
+
+  Widget _widgetAddButton() {
+    return ElevatedButton(
+      onPressed: submitForm,
+      child: Text(L10n.of(context).addTask),
+    );
+  }
+
+  Future<void> submitForm() async {
+    if (!_formKey.currentState!.validate()) return;
+    EasyLoading.show(status: L10n.of(context).addingTask);
+    final taskDraft = widget.taskList.taskBuilder();
+    taskDraft.title(_taskNameController.text);
+    if (_taskDescriptionController.text.isNotEmpty) {
+      taskDraft.descriptionText(_taskDescriptionController.text);
+    }
+    if (selectedDate != null) {
+      taskDraft.dueDate(
+        selectedDate!.year,
+        selectedDate!.month,
+        selectedDate!.day,
```

Comment:
these `!` shouldn't be necessary due to your check above.
```suggestion
        selectedDate.year,
        selectedDate.month,
        selectedDate.day,
```

Reply:
Need to keep it as it variable is global and giving syntax error.

![Screenshot 2024-06-05 at 6 36 50 PM](https://github.com/acterglobal/a3/assets/51526480/b42ba7f8-b2ce-4eec-af54-e50892614729)

Reply:
odd. looks like a false-positive of your IDE